added new scoring framework

gtsherman opened the pull request:
Note the package for the new framework is edu.gslis.scoring (as opposed
to edu.gslis.docscoring for the existing scorers) to allow us to
maintain the old scorers for backwards compatibility but keep the new
scorers distinct.

Commit:
added new scoring framework

Note the package for the new framework is edu.gslis.scoring (as opposed
to edu.gslis.docscoring for the existing scorers) to allow us to
maintain the old scorers for backwards compatibility but keep the new
scorers distinct.

Commit:
added new scoring framework

Note the package for the new framework is edu.gslis.scoring (as opposed
to edu.gslis.docscoring for the existing scorers) to allow us to
maintain the old scorers for backwards compatibility but keep the new
scorers distinct.

gtsherman commented:
I can change any calls to the old `Scorer` classes to use these new `DocScorer` classes as well; there shouldn't be very many of them.

bodom0015 committed:
This is why I hate copy+paste ;) (#13)

bodom0015 committed:
Updated POM for Maven Central sync requirements

craig-willis committed:
Create LICENSE

bodom0015 committed:
Merge pull request #19 from bodom0015/NDS-975

NDS-975: Updated POM for Maven Central sync requirements

bodom0015 committed:
Added license/developer info to POM

bodom0015 committed:
Merge pull request #20 from bodom0015/master

NDS-975: Added license/developer info to POM

Commit:
added BM25DocScorer

Commit:
added BM25DocScorer

Commit:
Merge branch 'scoring-lib' of https://github.com/gtsherman/ir-tools-1 into scoring-lib

Conflicts:
	pom.xml

gtsherman commented:
The old `edu.gslis.docscoring.*` classes are almost exclusively used in `edu.gslis.demo.*`. Below are the references in ir-tools to each docscoring class:

- QueryDocScorer
  - edu.gslis.demo.RunBM25
  - edu.gslis.demo.RunFilter
  - edu.gslis.demo.RunPLM
  - edu.gslis.demo.RunScorer
  - all edu.gslis.docscoring.Scorer* classes
- Scorer
  - QueryDocScorer
  - edu.gslis.filtering.session.SimpleFilterSessionImpl
- ScorerBM25
  - none
- ScorerDirichlet
  - edu.gslis.demo.RunPLM
  - edu.gslis.indexes.IndexWrapperLuceneImpl
    - In a `main` method with hard coded file paths from @craig-willis. Probably not worth keeping.
- ScorerDirichletKL
  - none
- ScorerDummy
  - none
- ScorerNull
  - none
- ScorerPLM
  - edu.gslis.demo.RunPLM
- ScorerSMART
  - In its own `main` method, again with hard coded file paths from @craig-willis.

I don't think it's worth my effort to rewrite any of the above, and I think we should consider ditching the "demo" classes (and should definitely at least remove the hard coded file paths, if not completely remove the `main` methods from the others).

I'm therefore going to say this PR is ready for review.

Commit:
added guava to pom

Commit:
added DoNothingDocScorer

gtsherman commented:
Craig, for when you look at this: I did a rebase for the first time and am surprised to see Mike's and your commits in the list above. If this signals that I did something wrong, let me know and I'll try to fix it.

craig-willis commented:
I'm going to hold off on merging these until we get the 0.2.0 build finalized for bioCaddie.  If there's anything related to bioCaddie in this or the other two PRs, let me know. I've seen the problem before where my PR contains older commits.  Did you pull from uiucGSLIS:master and push to your branch first?    @bodom0015  may have some other pointers.

gtsherman commented:
There's nothing related to bioCaddie in any of my PRs. 

I believe I pulled from uiucGSLIS/master into gtsherman/master and then rebased gtsherman/scoring-lib on top of gtsherman/master.

For what it's worth, looking at the commits you guys have made since this PR, merging this will not overwrite anything incorrectly. If it really does re-apply your commits, which I hope won't be the case, it won't break anything, because all you've edited since this PR is run.sh, the README, and Stopper.java, none of which are affected by this PR. So I think it should be safe to merge this.

Commit:
RelevanceModelScorer doesn't have to be instantiated for each document

Previously, the RelevanceModelScorer had to be constructed with the
query likelihood. This change allows the query likelihood to be computed
for each document on the fly.
